File: src/test/java/de/zalando/zally/rules/LimitNumberOfResourcesRuleTest.java
package de.zalando.zally.rules;

import io.swagger.models.Swagger;
import io.swagger.parser.SwaggerParser;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * Unit test for limiting number of resources rule.
 */
public class LimitNumberOfResourcesRuleTest {

    private Swagger validSwagger = getFixture("fixtures/limitNumberOfResourcesValid.json");
    private Swagger invalidSwagger = getFixture("fixtures/limitNumberOfResourcesInvalid.json");

    @Test
    public void shouldReturnNoViolations() {
        assertThat(new LimitNumberOfResourcesRule().validate(validSwagger)).isEmpty();
    }

    @Test
    public void shouldReturnViolations() {
        assertThat(new LimitNumberOfResourcesRule().validate(invalidSwagger)).isNotEmpty();
    }

    private Swagger getFixture(String fixture) {
        return new SwaggerParser().read(fixture);
    }
}

File: server/src/test/java/de/zalando/zally/rules/LimitNumberOfResourcesRuleTest.kt
package de.zalando.zally.rules

import de.zalando.zally.getFixture
import de.zalando.zally.testConfig
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class LimitNumberOfResourcesRuleTest {

    @Test
    fun positiveCase() {
        val swagger = getFixture("limitNumberOfResourcesValid.json")
        assertThat(LimitNumberOfResourcesRule(testConfig).validate(swagger)).isNull()
    }

    @Test
    fun negativeCase() {
        val swagger = getFixture("limitNumberOfResourcesInvalid.json")
        val result = LimitNumberOfResourcesRule(testConfig).validate(swagger)!!
        assertThat(result.paths).hasSameElementsAs(listOf(
                "/items",
                "/items/{item_id}",
                "/items10",
                "/items3",
                "/items4",
                "/items5",
                "/items6",
                "/items7",
                "/items8",
                "/items9"))
    }
}
